jobs/build/ocp4/Jenkinsfile

```diff
@@ -128,36 +128,31 @@ node {
             lock("github-activity-lock-${params.BUILD_VERSION}") {
                 stage("initialize") { joblib.initialize() }
                 buildlib.assertBuildPermitted(doozerOpts)
-                try {
-                    stage("build RPMs") {
+                stage("build RPMs") {
+                    try {
                         joblib.stageBuildRpms()
+                    } catch (err) {
+                        currentBuild.result = 'FAILURE'
                     }
-                } catch (err) {
-                    currentBuild.result = 'FAILURE'
                 }
 
                 // if the automation is not frozen perform compose
                 // otherwise if automation is frozen but there
                 // are rpms in the build plan perform compose
                 // and announce on slack
 
-                if(buildlib.getAutomationState(doozerOpts) in ["no", "False"]){
-                    lock("compose-lock-${params.BUILD_VERSION}") {
-                        stage("build compose") { joblib.stageBuildCompose() }
-                    }
-                } else if(joblib.buildPlan.buildRpms){
-                    lock("compose-lock-${params.BUILD_VERSION}") {
-                        stage("build compose") {
+                stage("build compose") {
+                    if (buildlib.isBuildPermitted(doozerOpts) || joblib.buildPlan.buildRpms) {
```

Comment:
`getAutomationState` was used here due to the situation described in the comment; `isBuildPermitted` is more nuanced because it permits a human running the build to succeed during a freeze (but in that case we won't update the plashet unless we're building an RPM for it).

if you can think of a clearer way to explain this for future readers of the code, that'd be helpful. it's certainly a bit confusing.

Reply:
possibly this exception just goes away at some point since we won't have merge windows and anyway would only be likely to run this job manually for an assembly

Reply:
Right. I believe this is fixed now.